src/overlays/actors/ovl_En_Clear_Tag/z_en_clear_tag.c

```diff
@@ -61,25 +76,991 @@ static ColliderCylinderInit D_809D5C40 = {
+            }
+        }
+    }
+}
+
+/**
+ * Draws the Arwing effects.
+ * Each effect type is drawn before the next. The function will apply a material that applies to all effects of that
+ * type while drawing the first effect of that type.
+ */
+void EnClearTag_DrawEffects(GlobalContext* globalCtx) {
+    GlobalContext* globalCtx2;
+    GraphicsContext* gfxCtx;
+    u8 isMaterialApplied;
+    s16 i;
+    EnClearTagEffect* effect;
+    EnClearTagEffect* firstEffect;
+
+    globalCtx2 = globalCtx;
+    gfxCtx = globalCtx2->state.gfxCtx;
+
+    firstEffect = effect = (EnClearTagEffect*)globalCtx2->specialEffects;
+    isMaterialApplied = 0;
+
+    OPEN_DISPS(gfxCtx, "../z_en_clear_tag.c", 1288);
+    func_80093D18(globalCtx2->state.gfxCtx);
+    func_80093D84(globalCtx2->state.gfxCtx);
+
+    // Draw all Debris effects.
+    for (i = 0; i < CLEAR_TAG_EFFECT_MAX_COUNT; i++, effect++) {
+        if (effect->type == CLEAR_TAG_EFFECT_DEBRIS) {
+            // Apply the debris effect material if it has not already been applied.
+            if (!isMaterialApplied) {
+                isMaterialApplied++;
+                gSPDisplayList(POLY_OPA_DISP++, gDebrisEffectMaterialDL);
+            }
+
+            // Draw the debris effect.
+            Matrix_Translate(effect->position.x, effect->position.y, effect->position.z, MTXMODE_NEW);
+            Matrix_Scale(effect->scale, effect->scale, effect->scale, MTXMODE_APPLY);
+            Matrix_RotateY(effect->rotationY, MTXMODE_APPLY);
+            Matrix_RotateX(effect->rotationX, MTXMODE_APPLY);
+            gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(gfxCtx, "../z_en_clear_tag.c", 1307),
+                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+            gSPDisplayList(POLY_OPA_DISP++, gDebrisEffectDL);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Clear_Tag/func_809D3798.s")
+    // Draw all ground flash effects.
+    effect = firstEffect;
+    isMaterialApplied = 0;
+    for (i = 0; i < CLEAR_TAG_EFFECT_MAX_COUNT; i++, effect++) {
+        if (effect->type == CLEAR_TAG_EFFECT_FLASH) {
+            // Apply the flash ground effect material if it has not already been applied.
+            if (!isMaterialApplied) {
+                gDPPipeSync(POLY_XLU_DISP++);
+                gDPSetEnvColor(POLY_XLU_DISP++, 255, 255, 200, 0);
+                isMaterialApplied++;
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Clear_Tag/func_809D38BC.s")
+            // Draw the ground flash effect.
+            gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 255, 255, 200, (s8)effect->primColor.a);
+            Matrix_Translate(effect->position.x, effect->floorHeight, effect->position.z, MTXMODE_NEW);
+            Matrix_RotateX(effect->floorTangent.x, MTXMODE_APPLY);
+            Matrix_RotateZ(effect->floorTangent.z, MTXMODE_APPLY);
+            Matrix_Scale(effect->scale + effect->scale, 1.0f, effect->scale + effect->scale, MTXMODE_APPLY);
```

Comment:
I think these should most likely be `2.0f *` (or just `2 *`)
```suggestion
            Matrix_Scale(2.0f * effect->scale, 1.0f, 2.0f * effect->scale, MTXMODE_APPLY);
```

Reply:
ye good catch here. Thanks!